Alfie/Alfie/Views/ProductDetails/ProductDetailsViewModel.swift

```diff
@@ -87,16 +88,46 @@ final class ProductDetailsViewModel: ProductDetailsViewModelProtocol {
         product?.priceType
     }
 
-    init(productId: String, product: Product?, dependencies: ProductDetailsDependencyContainer) {
+    var isAddToBagEnabled: Bool {
+        productHasStock
+        && hasColorSelected
+        && hasSizeSelected
+    }
+
+    private var hasColorSelected: Bool {
+        colorSelectionConfiguration.items.count > 1 ? selectedVariant?.colour != nil : true
```

Comment:
This condition could be improved:
```suggestion
        !colorSelection.configuration.items.isEmpty && selectedVariant?.colour != nil
```

Reply:
`isEmpty` is different than `count > 1`, so we should proceed with care here